src/main/scala/mesosphere/marathon/core/health/impl/MarathonHealthCheckManager.scala

```diff
@@ -112,13 +126,26 @@ class MarathonHealthCheckManager(
       }
     }
 
-  override def reconcileWith(appId: PathId): Future[Unit] =
-    appRepository.get(appId) flatMap {
+  override def reconcileWith(appId: PathId): Future[Unit] = {
+    def groupTasksByVersion(tasks: Iterable[Task]): Map[Timestamp, List[Task]] =
+      tasks.foldLeft(Map[Timestamp, List[Task]]()) {
+        case (acc, task) =>
+          task.launched match {
+            case Some(launched) =>
+              val version = launched.runSpecVersion
+              acc + (version -> (task :: acc.getOrElse(version, Nil)))
+            case None => acc
+          }
+      }
+
+    appRepository.get(appId).flatMap {
       case None => Future(())
       case Some(app) =>
         log.info(s"reconcile [$appId] with latest version [${app.version}]")
 
         val tasks: Iterable[Task] = taskTracker.appTasksSync(app.id)
+        val tasksByVersion = groupTasksByVersion(tasks)
```

Comment:
I think that we do need it, because Marathon starts a HealthCheckActor per app version with active tasks, see line #179 (https://github.com/mesosphere/marathon/pull/4311/files#diff-82c63be54fc7bbda204fdfde44104e48R179).